src/components/primitives/Button/Button.tsx

```diff
@@ -106,13 +123,26 @@ const Button = (
                   ...focusTextProps,
                   ...focusVisibleTextProps,
                   ...pressedTextProps,
+                  ...loadingTextProps,
                   ...disabledTextProps,
                 }}
               >
                 {isLoading && isLoadingText ? isLoadingText : children}
               </Box>
             ) : null}
+
             {endIcon && !isLoading ? endIcon : null}
+            {isLoading && spinnerPlacement === 'end' ? (
+              spinner ? (
+                spinner
+              ) : (
+                <Spinner
```

Comment:
Let's follow DRY for this Spinner element. We can keep this in a variable and use the same for start and end both.